cmd/api/src/api/v2/auth/saml.go

```diff
@@ -101,25 +215,26 @@ func (s ManagementResource) ServeMetadata(response http.ResponseWriter, request
 func (s ManagementResource) SAMLLoginHandler(response http.ResponseWriter, request *http.Request, ssoProvider model.SSOProvider) {
 	if ssoProvider.SAMLProvider == nil {
 		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusNotFound, api.ErrorResponseDetailsResourceNotFound, request), response)
-	} else if serviceProvider, err := serviceProviderFactory(request.Context(), s.config, *ssoProvider.SAMLProvider); err != nil {
+	} else if serviceProvider, err := auth.NewServiceProvider(*ctx.Get(request.Context()).Host, s.config, *ssoProvider.SAMLProvider); err != nil {
 		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, err.Error(), request), response)
 	} else {
-		providerResource := saml2.NewProviderResource(s.db, s.config, serviceProvider, samlWriteAPIErrorResponse)
-		binding, bindingLocation := providerResource.BindingTypeAndLocation()
-		// relayState is limited to 80 bytes but also must be integrity protected.
-		// this means that we cannot use a JWT because it is way too long. Instead,
-		// we set a signed cookie that encodes the original URL which we'll check
-		// against the SAML response when we get it.
-		if authReq, err := serviceProvider.MakeAuthenticationRequest(bindingLocation, binding, providerResource.ResponseBindingType); err != nil {
+		var (
+			binding         = saml.HTTPRedirectBinding
+			bindingLocation = serviceProvider.GetSSOBindingLocation(binding)
+		)
+		if bindingLocation == "" {
+			binding = saml.HTTPPostBinding
+			bindingLocation = serviceProvider.GetSSOBindingLocation(binding)
+		}
+
+		// TODO: add actual relay state support
```

Comment:
Why was the relay state support taken out? I see that it'll be added back in again, and I read the old code explaining a bit of what it means. But I still don't understand the why though

Reply:
Because it never actually functioned. The existing logic hardcoded idp initiated to true which negated all of the actual relay state support.

<img width="456" alt="image" src="https://github.com/user-attachments/assets/fe73c23a-fe3e-4fdf-8713-95880002dcd0">

Reply:
Ooh wow, that's def a good reason. Thanks for the explanation!